src/Model/Request/Request.php

```diff
@@ -73,6 +74,8 @@ public function __construct(
 
         $this->headers = new KeyValueCollection();
         $this->headers->set('Content-Type', JsonApiInterface::CONTENT_TYPE);
+
+        $this->customQueryParameters = new KeyValueCollection();
```

Comment:
Please add the functionality here in the constructor, that not related query parameters in the URI will automatically be added to the custom query parameters.